### ISSUE TITLE
Require up.json for deploys

### ISSUE THREAD

**Issue by tj** [apex/up#386]
It's a nice little gimmick to allow `up` with only say one `app.js` file, but there's not much practical value, it might be nicer to just error if `up.json` isn't present so it doesn't try deploying random directories by accident.

- [x] require a `.name`
- [x] make up.json required
- [x] maybe prompt asking to create up.json for them with the dirname (or enter a name)
- [x] maybe infer/prompt for a few other fields ("type" == "static")
- [ ] remove gimmicky examples
- [ ] adjust documentation

**Comment by stephenmathieson** [comment]
I agree 100%.

Past discussion: #191

**Comment by fullstacknz** [comment]
I came here to add this feature request if it hadn't already been made. I definitely vote for this to be included.

I recently used the 'up production' command in the wrong project and had to sit and wait for a full stack deployment (on a slow connection) then delete the stack again straight after. I noticed my mistake quickly but this would have caused me a bit of confusion down the track if not.

Allowing the 'up' command to work without the up.json file is not ideal. Established command line tools such as npm prevent this kind of mistake.
